x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +870,194 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+
+	// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+	clPoolId1 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 1
+	clPoolId2 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 2
+
+	epochInfo := s.App.IncentivesKeeper.GetEpochInfo(s.Ctx)
+
+	// 3. Create external no-lock gauges for CL pools
+	_, clPoolExternalGaugeIdPool1, clPoolExternalGaugeIdPool2 := s.FunctionalHelperCreateExternalGauges(clPoolId1.GetId(), clPoolId2.GetId())
+
+	// 4. Create Distribution records to incentivize internal CL no-lock gauges
+	s.FunctionalHelperIncentivizeInternalGauge(clPoolId1.GetId(), clPoolId2.GetId(), epochInfo.Duration, false)
+
+	// 5. let epoch 1 pass
+	// ******************** EPOCH 1 *********************
+	// let 1 epoch pass
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.GetIdentifier(), 1)
+
+	// Validate the results
+	clPoolInternalGaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId1.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 1, sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(1_000_000)), sdk.NewCoin("usdc", sdk.NewInt(1_000_000))))
+
+	internalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
+	s.Require().NoError(err)
+
+	// check  internalIncentiveCLPool
+	s.Require().Equal(2, len(internalIncentiveClPool))
+	emissionRate := sdk.NewDecFromInt(sdk.NewInt(1_000_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), "eth", module.GetAddress().String(), emissionRate, s.Ctx.BlockTime(), sdk.NewInt(1_000_000), internalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), "usdc", module.GetAddress().String(), emissionRate, s.Ctx.BlockTime(), sdk.NewInt(1_000_000), internalIncentiveClPool[1])
+
+	// check  externalIncentiveCLPool
+	s.Require().Equal(2, len(externalIncentiveClPool))
+	emissionRate = sdk.NewDecFromInt(sdk.NewInt(1_000_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds() * 2).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), "eth", module.GetAddress().String(), emissionRate, s.Ctx.BlockTime(), sdk.NewInt(500_000), externalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), "usdc", module.GetAddress().String(), emissionRate, s.Ctx.BlockTime(), sdk.NewInt(500_000), externalIncentiveClPool[1])
+
+	// 6. Remove distribution records for internal incentives using HandleReplacePoolIncentivesProposal
+	s.FunctionalHelperIncentivizeInternalGauge(clPoolId1.GetId(), clPoolId2.GetId(), epochInfo.Duration, true)
+
+	// 7. let epoch 2 pass
```

Comment:
note: i believe we distribtue internal incentives starting 2nd epoch because in 2nd epoch we make those internal gauges active with coins to distributed in the struct